### DISCUSSION TITLE
Use the packaged file without any error message

### DISCUSSION THREAD

**Discussion by zfy108** [pyinstaller/pyinstaller#6421] [Help]
When I used pyinstaller to package the exe file, I found that the following error message popped up on another computer,“[15380] Failed to execute script '***' due to unhandled exception!” and there was no way to locate the cause of the problem.
I tried many ways but no other information was printed includes that “debug” and “console”
I want to know if there is any way to know what caused it
any infomation will be appreciated.

**Comment by rokm** [comment]
Is this a `console` or `noconsole`/`windowed` build? I.e., do you get that `“[15380] Failed to execute script '***' due to unhandled exception!” ` message in a console or in a popup window? The preceding process ID (`“[15380]"`) makes me think this is a console build, but in that case you should also see the preceding error strack trace. Is that not the case?

But in the end, if you really cannot see any other information, I suppose you can implement proper top-level error handling, catch the exception, and log its information to a file.

**Reply by zfy108** [reply]
this is a console build.
I want to know how to implement proper top-level error handling and catch the exception. becase my code have not run
any infomation will be appreciated.

**Reply by rokm** [reply]
> this is a console build.

And there is no other output before the `“[15380] Failed to execute script '***' due to unhandled exception!”` line? In `console` build, there should be a traceback printed (by python interpreter itself) before we print that final error message...

By the way, is the `'***'` name of your script, or is it one of PyInstaller's run-time hooks (`pyi_rth_*`)?

> I want to know how to implement proper top-level error handling and catch the exception.

Put all your code into one global `try`/`except` block and handle the error (i.e., print it to console or save it to a file) there. That's assuming the error comes from your code and not from PyInstaller's run-time hooks...
